jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,34 +5,79 @@
  * Copyright (c) 2024, Gluu, Inc.
  */
 
-use std::collections::HashMap;
-
 use super::cedar_schema::CedarSchema;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt;
 
 /// PolicyStoreMap it is a store for `PolicyStore` accessible by key.
-#[derive(Debug, serde::Deserialize)]
+#[derive(Debug, Deserialize)]
 pub(crate) struct PolicyStoreMap {
     #[serde(flatten)]
     pub policy_stores: HashMap<String, PolicyStore>,
 }
 
 /// PolicyStore contains all the data the Cedarling needs to verify JWT tokens and evaluate policies
-#[derive(Debug, Clone, serde::Deserialize)]
+#[derive(Debug, Clone, Deserialize)]
 pub struct PolicyStore {
     pub schema: CedarSchema,
     #[serde(deserialize_with = "deserialize_policy_set::parse")]
     pub policies: cedar_policy::PolicySet,
+    #[allow(dead_code)]
+    pub trusted_issuers: Option<Vec<TrustedIssuer>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TrustedIssuer {
+    pub name: String,
+    pub description: String,
+    pub openid_configuration_endpoint: String,
+    pub token_metadata: Option<Vec<TokenMetadata>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TokenMetadata {
+    #[serde(
+        rename = "type",
+        deserialize_with = "deserialize_policy_set::parse_token_kind"
```

Comment:
We need to use macros `deserialize_with` only if we can't implement `Deserialize`. It happens when we need to deserialize type from foreign crate.

For example look to the `CedarSchema` struct.

Reply:
i used the macro so i could make deserialization be case-insensitive. i think it's a better experience for the end-user.

Reply:
implemented `Deserialize` instead of `deserialize_with` here: [0b2128f](https://github.com/JanssenProject/jans/pull/9893/commits/0b2128fd1f067bfc7a3b5b6cd86d88d698a63b69)